client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -209,170 +203,145 @@ const useActiveAddOnsDefs = ( selectedSite: SiteDetails | null ) => {
+						siteFeatures?.active?.includes( WPCOM_FEATURES_AI_ASSISTANT )
+					) {
+						return false;
+					}
 
-				// if it's a storage add on
-				if ( addOn.productSlug === PRODUCT_1GB_SPACE ) {
-					// if storage add ons are not enabled in the config, remove them
-					if ( ! isStorageAddonEnabled() ) {
-						return null;
+					// TODO: Remove this check once paid WPCOM stats is live.
+					// gate the Jetpack Stats add-on on a feature flag
+					if (
+						[ PRODUCT_JETPACK_STATS_PWYW_YEARLY, PRODUCT_JETPACK_STATS_YEARLY ].includes(
+							addOn.productSlug
+						) &&
+						! config.isEnabled( 'stats/paid-wpcom-stats' )
+					) {
+						return false;
 					}
 
-					// if storage add on hasn't loaded yet
-					if ( isLoadingBillingTransactions || ! product ) {
-						return {
-							...addOn,
-							name,
-							description,
-							isLoading: isLoadingBillingTransactions,
-						};
+					// Hide Stats Personal add-on if the site is marked as commercial.
+					if (
+						isSiteMarkedCommercial === true &&
+						PRODUCT_JETPACK_STATS_PWYW_YEARLY === addOn.productSlug
+					) {
+						return false;
 					}
 
-					// if storage add on is already purchased
+					// Hide Stats Commercial add-on if the site is not marked as commercial.
 					if (
-						spaceUpgradesPurchased.findIndex(
-							( spaceUpgrade: any ) => spaceUpgrade === addOn.quantity
-						) >= 0 &&
-						product
+						isSiteMarkedCommercial === false &&
+						PRODUCT_JETPACK_STATS_YEARLY === addOn.productSlug
 					) {
-						return {
-							...addOn,
-							name,
-							description,
-							purchased: true,
-						};
+						return false;
 					}
 
-					const currentMaxStorage = mediaStorage?.max_storage_bytes / Math.pow( 1024, 3 );
-					const availableStorageUpgrade = STORAGE_LIMIT - currentMaxStorage;
+					// TODO: Show the Stats Commercial add-on for a commercial site that has purchased the Personal plan.
 
-					// if the current storage add on option is greater than the available upgrade
-					if ( ( addOn.quantity ?? 0 ) > availableStorageUpgrade ) {
-						return {
-							...addOn,
-							name,
-							description,
-							exceedsSiteStorageLimits: true,
-						};
+					// remove Jetpack Stats add-on if the site already has a paid stats feature through a paid plan.
+					if (
+						[ PRODUCT_JETPACK_STATS_PWYW_YEARLY, PRODUCT_JETPACK_STATS_YEARLY ].includes(
+							addOn.productSlug
+						) &&
+						siteFeatures?.active?.includes( FEATURE_STATS_PAID )
+					) {
+						return false;
 					}
-				}
 
-				if ( ! product ) {
-					// will not render anything if product not fetched from API
-					// probably need some sort of placeholder in the add-ons page instead
-					return null;
-				}
+					return true;
+				} )
+				.map( ( addOn ) => {
+					const product = productsList[ addOn.productSlug ];
+					const name = addOn.name ? addOn.name : product.product_name;
+					const description = addOn.description ?? product.description;
```

Comment:
This was the main change to apply `useMemo` instead of the more state-bound `createSelector` memoizer @jeyip

Reply:
Double checked the `getProductBySlug`, `getProductName`, and `getProductDescription` selectors we replaced. The updates look good to me 👍